### PR TITLE
Fix jail-online-time with offline players

### LINKED ISSUE #5658: Jailing offline players doesn't work with jail-online-time set to true
### Type of bug

Other unexpected behaviour

### `/ess dump all` output

n/a

### Error log (if applicable)

n/a

### Bug description

Example player data file with `jail-online-time` option set to `true` after jailing:

the command is `/jail {player} {jail} 3m`

when player is online:
```
  jail: 180000
  onlinejail: 55718738
```

when player is offline:
```
  jail: 180000
  onlinejail: 3600
```

https://github.com/EssentialsX/Essentials/blob/2.x/Essentials/src/main/java/com/earth2me/essentials/commands/Commandtogglejail.java#L85

My guess is that `player.getBase().getStatistic(PLAY_ONE_TICK)` returns some weird value (maybe 0) for offline players and that's the thing that needs to be fixed, however I'm not sure about the correct approach to that.

[(Discord forum)](https://discord.com/channels/390942438061113344/1200986919119638640/1203504708187070465)

### Steps to reproduce

Above ^

### Expected behaviour

Time should be set correctly lol

### Actual behaviour

Time isn't set correctly and player gets unjailed instantly on login

### PR DESCRIPTION
**Pull request by pop4959**
Fixes #5658

Apparently you cannot get statistics for players returned by getPlayer when they are offline. This may be an upstream bug, but regardless, an easy fix here is to simply allow getting an OfflinePlayer reference from the User base for this purpose.

Below is a demonstration of the issue.

```java
sender.sendMessage(String.valueOf(player.getBase().getStatistic(PLAY_ONE_TICK)));
sender.sendMessage(String.valueOf(ess.getServer().getOfflinePlayer(player.getBase().getUniqueId()).getStatistic(PLAY_ONE_TICK)));
sender.sendMessage(String.valueOf(timeDiff / 50));
```

90fd0a3c-20fb-42e3-badf-372405fc50b6

As you can see, the offline player returns the correct result, but the player base does not.

### CODE REVIEW

File: Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java [hunk 1]
```diff
@@ -31,4 +32,8 @@ public World getWorld() {
     public Location getLocation() {
         return base.getLocation();
     }
+
+    public OfflinePlayer getOffline() {
+        return base.getServer().getOfflinePlayer(base.getUniqueId());
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java]: Does this do a profile lookup if a user isn't in the server's cache? We want to be wary about slow lookups, especially for Bedrock/Vault NPC UUIDs.

Comment (pop4959) [reply Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java]: Mentioned in Discord - but we can potentially skip the lookup by just checking if the base is an instance of `OfflinePlayerStub` and if so, get its base which is itself an OfflinePlayer.

Not sure if this is something we want to do VS just update usages of OfflinePlayer and implement the OfflinePlayer methods that are expected here.

**Comment by pop4959** [comment]
Not an upstream bug as initially thought, this is just some Essentials goofiness.

**Review by JRoy — DISMISSED** [review]
not a fan of this implementation, would rather a logoff-time entry be added to essentials userdata so we can do the calculation on our end. loading offline playerdata is unpredictable and can lead to even worse I/O issues that loading Essentials userdata.

**Comment by pop4959** [comment]
> not a fan of this implementation, would rather a logoff-time entry be added to essentials userdata so we can do the calculation on our end. loading offline playerdata is unpredictable and can lead to even worse I/O issues that loading Essentials userdata.

Yeah that works too, alternative would be to get the offline player from the offline player stub which has already done the lookup at that point (so it would incur no additional cost). Or actually implementing the method on the stub itself (which we may want to update anyway - it's horribly outdated at the moment).

File: Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java (hunk 1 above)
Comment (pop4959) [reply Essentials/src/main/java/com/earth2me/essentials/PlayerExtension.java]: Implemented as mentioned above. Should be "good enough" without having to do a lot of extra work to make this play nicer.

**Review by JRoy — APPROVED** [review]

**Pull request merged**